pkg/runtime/interfaces.go

```diff
@@ -30,12 +30,23 @@ const (
 	APIVersionInternal = "__internal"
 )
 
+// GroupVersioner conveys information about a desired target version for objects being converted.
+type GroupVersioner interface {
```

Comment:
In that case, maybe the scheme should give a list of options and this interface chooses one.